src/groupByKey.ts

```diff
@@ -2,6 +2,21 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T extends object>(
+  items: T[],
+  key: keyof T,
+): GroupsMap<T> {
+  const result: GroupsMap<T> = {};
+
+  items.forEach((element) => {
+    const stringKey = String(element[key]);
+
+    if (Object.hasOwn(result, stringKey)) {
```

Comment:
The method `Object.hasOwn` is not a standard JavaScript method. You should use `Object.prototype.hasOwnProperty.call(result, stringKey)` to check if the object has a property with the given key.